RockWeb/Blocks/Groups/GroupDetail.ascx.cs

```diff
@@ -825,6 +825,102 @@ protected void btnCancel_Click( object sender, EventArgs e )
+            if ( group != null )
+            {
+                // clone the group
+                var newGroup = group.Clone( false );
+                newGroup.CreatedByPersonAlias = null;
+                newGroup.CreatedByPersonAliasId = null;
+                newGroup.CreatedDateTime = RockDateTime.Now;
+                newGroup.ModifiedByPersonAlias = null;
+                newGroup.ModifiedByPersonAliasId = null;
+                newGroup.ModifiedDateTime = RockDateTime.Now;
+                newGroup.Id = 0;
+                newGroup.Guid = Guid.NewGuid();
+                newGroup.IsSystem = false;
+                newGroup.Name = group.Name + " - Copy";
+
+                var auths = authService.GetAuths( group.Id );
+                rockContext.WrapTransaction( () =>
+                {
+                    groupService.Add( newGroup );
+                    rockContext.SaveChanges();
+
+                    newGroup.LoadAttributes( rockContext );
+                    if ( group.Attributes != null && group.Attributes.Any() )
+                    {
+                        foreach ( var attributeKey in group.Attributes.Select( a => a.Key ) )
+                        {
+                            string value = group.GetAttributeValue( attributeKey );
+                            Guid guidValue = value.AsGuid();
+                            newGroup.SetAttributeValue( attributeKey, value );
+                        }
+                    }
+                    newGroup.SaveAttributeValues( rockContext );
+
+                    /* Take care of Group Member Attributes */
+                    var entityTypeId = EntityTypeCache.Read( typeof( GroupMember ) ).Id;
+                    string qualifierColumn = "GroupId";
+                    string qualifierValue = group.Id.ToString();
+                    // Get the existing attributes for this entity type and qualifier value
+                    var attributes = attributeService.Get( entityTypeId, qualifierColumn, qualifierValue );
+
+                    foreach ( var attribute in attributes )
+                    {
+                        var newAttribute = attribute.Clone( false );
+                        newAttribute.Id = 0;
+                        newAttribute.Guid = Guid.NewGuid();
+                        newAttribute.IsSystem = false;
+
+                        foreach ( var qualifier in attribute.AttributeQualifiers )
+                        {
+                            var newQualifier = qualifier.Clone( false );
+                            newQualifier.Id = 0;
+                            newQualifier.Guid = Guid.NewGuid();
+                            newQualifier.IsSystem = false;
+                            newAttribute.AttributeQualifiers.Add( qualifier );
+                        }
+
+                        Rock.Attribute.Helper.SaveAttributeEdits( newAttribute, entityTypeId, qualifierColumn, newGroup.Id.ToString(), rockContext );
```

Comment:
Using the Helper.SaveAttributeEdits isn't really needed here. You should just be able to use something like `attributeValueService.Add( newAttribute );` and then call `rockContext.SaveChanges()` again to save the new values.